dotnet/Utilities/IResourceLoader.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Ara3D.DotNetUtilities
+{
+    public interface IResourceLoader
```

Comment:
Okay, so now I get it. This is an asynchronous (task-based) resource loader, like they use in Three.JS (and a lot of JavaScript projects). I like it a lot! I'll go through with a fine tooth comb tomorrow, because my brain isn't much good today.